src/main/webapp/app/tasks/local-task.service.ts

```diff
@@ -1,37 +1,55 @@
-import { Injectable } from '@angular/core';
-import { Observable, of } from 'rxjs';
-import { v4 as uuid } from 'uuid';
+import { Injectable } from "@angular/core";
+import { Observable, of } from "rxjs";
+import { v4 as uuid } from "uuid";
 
-import { Task } from './task';
-import { TaskService } from './task.service';
+import { Task } from "./task";
+import { TaskService } from "./task.service";
 
 @Injectable()
 export class LocalTaskService implements TaskService {
-
-  private static readonly STORAGE_KEY: string = 'tiny.tasks';
+  private static readonly STORAGE_KEY: string = "tiny.tasks";
 
   getAll(): Observable<Task[]> {
     return of(this.readTasks());
   }
 
   create(name: string): Observable<Task> {
     const tasks = this.readTasks();
-    const task = {id: uuid(), name};
+    const task = { id: uuid(), name, status: "Started" };
     tasks.push(task);
     this.writeTasks(tasks);
     return of(task);
   }
 
   delete(id: string): Observable<void> {
     const tasks = this.readTasks();
-    const index = tasks.findIndex(task => task.id === id);
+    const index = tasks.findIndex((task) => task.id === id);
     if (index !== -1) {
       tasks.splice(index, 1);
       this.writeTasks(tasks);
     }
     return of(null);
   }
 
+  toggleStatus(id: string): Observable<void> {
+    const tasks = this.readTasks();
+    tasks.forEach((task) =>
+      task.id === id
+        ? task.status === "Started"
+          ? (task.status = "Done")
+          : (task.status = "Started")
+        : null
```

Comment:
this is pretty hard to understand, in my opinion for each and then match the id and do the assignment is not a very good idea, you would rather find the index and set it directly or delegate to a method or there are lot of different other options to choose from, but two nested ternary operators?